atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/SweepCommand.java

```diff
@@ -0,0 +1,170 @@
+
+        if ((table != null)) {
+            Optional<byte[]> startRow = Optional.of(new byte[0]);
+            if (row != null) {
+                startRow = Optional.of(decodeStartRow(row));
+            }
+            tableToStartRow.put(table, startRow);
+        } else if (namespace != null) {
+            Set<String> tablesInNamespace = Sets.filter(services.getKeyValueService().getAllTableNames(),
+                    Predicates.containsPattern("^" + namespace + "\\."));
+            for (String table : tablesInNamespace) {
+                tableToStartRow.put(table, Optional.of(new byte[0]));
+            }
+        } else if (sweepAllTables) {
+            tableToStartRow.putAll(
+                    Maps.asMap(
+                            Sets.difference(services.getKeyValueService().getAllTableNames(), AtlasDbConstants.hiddenTables),
+                            Functions.constant(Optional.of(new byte[0]))));
+        }
+
+        for (Map.Entry<String, Optional<byte[]>> entry : tableToStartRow.entrySet()) {
+            final String table = entry.getKey();
+            Optional<byte[]> startRow = entry.getValue();
+
+            final AtomicLong cellsExamined = new AtomicLong();
+            final AtomicLong cellsDeleted = new AtomicLong();
+
+            while (startRow.isPresent()) {
+                Stopwatch watch = Stopwatch.createStarted();
+                SweepResults results = sweepRunner.run(table, config.getSweepBatchSize(), startRow.get());
+                System.out.println(String.format("Swept from %s to %s in table %s in %d ms, examined %d unique cells, deleted %d cells.",
+                        encodeStartRow(startRow), encodeEndRow(results.getNextStartRow()),
+                        table, watch.elapsed(TimeUnit.MILLISECONDS),
+                        results.getCellsExamined(), results.getCellsDeleted()));
+                startRow = results.getNextStartRow();
+                cellsDeleted.addAndGet(results.getCellsDeleted());
+                cellsExamined.addAndGet(results.getCellsExamined());
+                try {
+                    Thread.sleep(config.getSweepPauseMillis());
+                } catch (InterruptedException e) {
+                    throw Throwables.rewrapAndThrowUncheckedException(e);
+                }
+            }
+
+            services.getTransactionManager().runTaskWithRetry(new TxTask() {
+                @Override
+                public Void execute(Transaction t) {
+                    SweepPriorityTable priorityTable = SweepTableFactory.of().getSweepPriorityTable(t);
+                    SweepPriorityTable.SweepPriorityRow row = SweepPriorityTable.SweepPriorityRow.of(table);
+                    priorityTable.putWriteCount(row, 0L);
+                    priorityTable.putCellsDeleted(row, cellsDeleted.get());
+                    priorityTable.putCellsExamined(row, cellsExamined.get());
+                    priorityTable.putLastSweepTime(row, System.currentTimeMillis());
```

Comment:
It's supposed to be wallclock time here; the logic here for the background sweeper is "I just swept this and it hasn't received very many writes since then, I don't have to sweep it again all that soon".

I have a change on a PR somewhere that adds an additional field that is 'the lowest immutable ts I was using when I swept the entirety of this table' , which is only necessary for the implementation of another task that is capable of cleaning up old transactions.